csrc/device_lower/pass/circular_buffer.cpp

```diff
@@ -705,6 +862,38 @@ class ClonePipelinedTmaCircularBufferLoopAndInsertSync
     return wait_exprs;
   }
 
+  // For each mbarrier that is used to wait for the finish reading of circular
+  // buffers in the given loop, find all the expressions that use the circular
+  // buffer tensor tracked by this mbarrier.
+  std::unordered_map<TensorView*, std::unordered_set<Expr*>>
+  getAllWarMbarriersToUses() {
```

Comment:
Do we have a test where a WAR mbarrier maps to multiple expressions?

Reply:
We are already mapping all expressions to a single mbarrier, so yes we do.

But we don't have any test where we are using different mbarriers for different expressions.